### ISSUE TITLE
Could not upgrade to LSP4E 0.10.0

### ISSUE THREAD

**Issue by norru** [eclipse/corrosion#250]
Getting the following error after my last attempt to upgrade from snapshot:

```
Your original request has been modified.
  "Language Server Protocol client for Eclipse IDE (Incubation)" is already installed, so an update will be performed instead.
Cannot complete the install because one or more required items could not be found.
  Software being installed: Language Server Protocol client for Eclipse IDE (Incubation) 0.10.0.201905140816 (org.eclipse.lsp4e 0.10.0.201905140816)
  Missing requirement: Language Server Protocol client for Eclipse IDE (Incubation) 0.10.0.201905140816 (org.eclipse.lsp4e 0.10.0.201905140816) requires 'osgi.bundle; org.eclipse.ui.workbench 3.115.0' but it could not be found
```

**Comment by mickaelistria** [comment]
LSP4E 0.10.0 requires newer Eclipse Platform, that will be released in ~1 month.
You can in the meantime either stay with 0.9, or start adopting newer Eclipse Platform from http://download.eclipse.org/eclipse/downloads/drops4/

**Comment by mickaelistria** [comment]
Closing as there is not much that can be done on that matter.

**Comment by norru** [comment]
Sounds good. What is the update site?

**Comment by mickaelistria** [comment]
https://download.eclipse.org/eclipse/updates/4.12-I-builds/ would be even better, however if you started from an EPP package, update may not work as EPP "locks" the version of the Eclipse Platform.

**Comment by norru** [comment]
With the "4.12milestones" I'm getting this:

![image](https://user-images.githubusercontent.com/889291/58021305-10931c80-7afa-11e9-8e35-275e8d75419d.png)

**Comment by norru** [comment]
The 4.12-I-builds does a similar thing:

![image](https://user-images.githubusercontent.com/889291/58021427-67005b00-7afa-11e9-853a-d38aefb2b710.png)

**Comment by mickaelistria** [comment]
Yes, that's the issue I mentioned. It's a bit hard to upgrade a package with a snapshot of Platform.
The easiest way is to start from a vanilla Eclipse SDK from https://download.eclipse.org/eclipse/downloads/drops4/I20190519-1800/ and to install Corrosion in it.
Or you can wait for next week, as the snapshots of the "Eclipse IDE for Rust package" should be updated with newer version of Platform, so you'll be able to upgrade it all without such conflict.

**Comment by norru** [comment]
Ah, ok, I'll wait until next week.

**Comment by norru** [comment]
Upgrading to Eclipse 2019-06 fixes this issue.

**Comment by norru** [comment]
I just got a similar issue with `LSP4E 0.13.0`. It looks like an identical issue. Will it be fixed by Eclipse 2020.03?

**Comment by mickaelistria** [comment]
I hope we'll be able to release LSP4E soon. After that, we can rebuild Corrosion with newer versions of LSP4E to hopefully avoid this issue.
